content/providers/05-observability/humanloop.mdx

```diff
@@ -0,0 +1,265 @@
+
+    ```ts highlight="3-13,19"
+    import { openai } from '@ai-sdk/openai';
+    import { generateText } from 'ai';
+    import { NodeSDK } from '@opentelemetry/sdk-node';
+    import { getNodeAutoInstrumentations } from '@opentelemetry/auto-instrumentations-node';
+    import dotenv from 'dotenv';
+
+    dotenv.config();
+
+    const sdk = new NodeSDK({
+      instrumentations: [getNodeAutoInstrumentations()],
+    });
+
+    sdk.start();
+
+    async function main() {
+      // ... Vercel AI SDK calls ...
+
+      // Must call shutdown to flush traces
+      await sdk.shutdown();
+    }
+
+    main().catch(console.error);
+    ```
+
+    Your calls to the AI SDK should now be logged to Humanloop.
+
+  </Tab>
+</Tabs>
+
+## Trace Grouping
+
+To group multiple AI SDK calls into a single Flow Log, create and pass a Flow Log ID to the telemetry metadata of each AI SDK call.
+
+1. Create a Flow Log in Humanloop
+2. Pass the Flow Log ID to each AI SDK call
+3. Update the Flow Log when all executions are complete
+
+The Flow Log serves as a parent container for all related Prompt Logs in Humanloop.
+
+```ts
+import { HumanloopClient } from 'humanloop';
+
+const humanloop = new HumanloopClient();
+
+async function main() {
+  const flow = await humanloop.flows.upsert({
+    path: 'Plethora of Poetry',
+    attributes: {},
+  });
+  const flowLog = await humanloop.flows.log({
+    id: flow.id,
+  });
+
+  const outputs = [];
+
+  for (const poetName of ['Edgar Allan Poe', 'Mary Shelley', 'Lord Byron']) {
+    const result = await generateText({
+      model: openai('gpt-3.5-turbo'),
+      maxTokens: 50,
+      prompt: `Write me a poem in the style of ${poetName}.`,
+      experimental_telemetry: {
+        isEnabled: true,
+        functionId: `poet-${poetName.toLowerCase().replace(' ', '-')}`,
+        metadata: {
+          humanloopFlowId: flowLog.id,
+          humanloopPromptPath: `Poets/${poetName}`,
+        },
+      },
+    });
+
+    outputs.push(result.text);
+  }
+
+  await humanloop.flows.updateLog(flowLog.id, {
+    traceStatus: 'complete',
+    output: outputs.join('\n\n'),
+  });
+
+  await sdk.shutdown();
+}
+```
+
+## Debugging
+
+If you aren't using Next.js 15+, you will also need to enable the experimental instrumentation hook (available in 13.4+).
+
+```javascript filename="next.config.js"
+module.exports = {
+  experimental: {
+    instrumentationHook: true,
+  },
+};
+```
+
+## Resources
+
+To see a full example of instrumenting your application, check out the Humanloop [AI SDK Guides](https://humanloop.com/docs/v5/vercel-ai-sdk).
+
+After instrumenting your AI SDK application with Humanloop, you can then:
+
+- Experiment with different [versions of Prompts](https://humanloop.com/docs/v5/guides/evals/comparing-prompts) and try them out in the Editor
```

Comment:
"Try tweaking your Prompt in the workspace editor to improve its performance"

Reply:
I see your point about provider: they can tweak the Prompt but not call it after optimisation. Let's mention that briefly in the same paragraph: "Our AI SDK Provider implementation is coming soon, allowing you to switch between Prompt versions as you make tweaks"